pkg/blobinfocache/internal/prioritize/prioritize.go

```diff
@@ -78,6 +90,34 @@ type CandidateWithTime struct {
 	LastSeen  time.Time                              // Time the candidate was last known to exist (either read or written) (not set for Candidate.UnknownLocation)
 }
 
+// CandidateWithLocation returns a complete CandidateWithTime combining (template from CandidateTemplateWithCompression, location, lastSeen)
+func (template CandidateTemplate) CandidateWithLocation(location types.BICLocationReference, lastSeen time.Time) CandidateWithTime {
+	return CandidateWithTime{
+		Candidate: blobinfocache.BICReplacementCandidate2{
+			Digest:               template.digest,
+			CompressionOperation: template.compressionOperation,
+			CompressionAlgorithm: template.compressionAlgorithm,
+			UnknownLocation:      false,
+			Location:             location,
+		},
+		LastSeen: lastSeen,
+	}
+}
+
+// CandidateWithUnknownLocation returns a complete CandidateWithTime for a template from CandidateTemplateWithCompression and an unknown location.
+func (template CandidateTemplate) CandidateWithUnknownLocation() CandidateWithTime {
+	return CandidateWithTime{
+		Candidate: blobinfocache.BICReplacementCandidate2{
+			Digest:               template.digest,
+			CompressionOperation: template.compressionOperation,
+			CompressionAlgorithm: template.compressionAlgorithm,
+			UnknownLocation:      true,
+			Location:             types.BICLocationReference{Opaque: ""},
+		},
+		LastSeen: time.Time{},
```

Comment:
nit: these initializations may be omitted.

Reply:
I prefer being explicit here — especially about the fact that `Time` is not set.